### ISSUE TITLE
MemoryError: memory allocation failed, allocating 512 bytes

### ISSUE THREAD

**Issue by mattzzw** [mattzzw/obi_socket#7]
Memory seems to become a limited resource...
```
ERROR:picoweb:596484708.786 <HTTPRequest object at 3fff5c60> <StreamWriter <socket state=2 timeout=0 incoming=0 off=0>> MemoryError('memory allocation failed, allocating 512 bytes',)
Traceback (most recent call last):
  File "picoweb/__init__.py", line 203, in _handle
  File "obi_socket.py", line 129, in index
  File "config.py", line 47, in load
MemoryError: memory allocation failed, allocating 512 bytes
```

**Comment by mattzzw** [comment]
https://docs.micropython.org/en/latest/reference/constrained.html

**Comment by mattzzw** [comment]
Fragmented heap is a bitch. But my code was/is bad as well.
Partially fixed but still some work to do:
- Multiple import of same module in different files (especially config.py)
- Using a pre-allocated buffer helps a lot
- Better partitioning of code

**Comment by mattzzw** [comment]
Still an issue:

```
INFO:picoweb:596737575.531 <HTTPRequest object at 3fff70f0> <StreamWriter <socket state=2 timeout=0 incoming=3fff91a8 off=22>> "POST /setup"
DEBUG: Before:  4288
ERROR:picoweb:596737575.531 <HTTPRequest object at 3fff70f0> <StreamWriter <socket state=2 timeout=0 incoming=3fff91c8 off=50>> MemoryError('memory allocation failed, allocating 278 bytes',)
Traceback (most recent call last):
  File "picoweb/__init__.py", line 203, in _handle
  File "obi_socket.py", line 147, in setup
  File "picoweb/__init__.py", line 72, in read_form_data
  File "uasyncio/__init__.py", line 104, in read
MemoryError: memory allocation failed, allocating 278 bytes
ERROR:picoweb:<StreamReader <socket state=3 timeout=0 incoming=0 off=0> <socket state=3 timeout=0 incoming=0 off=0>>: EOF on request start
```

**Comment by mattzzw** [comment]
Changed dict to list, split up the cfg forms and removed basically all debug print code to save memory.

Resolved by commit 33323c5b140bb1d9169b4bba74e589b614fbf60b and b58faa8e8a8b4eacffc5689ef7b036394e9e99bf
